test/test-setup.js

```diff
@@ -0,0 +1,89 @@
+"use strict";
+
+const test = require("sdk/test");
+const tabs = require("sdk/tabs");
+const {ActivityStreams} = require("lib/ActivityStreams");
+const httpd = require("./lib/httpd");
+const {doGetFile} = require("./lib/utils");
+const {before, after} = require("sdk/test/utils");
+
+const PORT = 8099;
+const path = "/dummy-activitystreams.html";
+const url = `http://localhost:${PORT}${path}`;
+const otherUrl = `http://localhost:${PORT}/dummy-other.html`;
+
+let srv;
+let app;
+let openTabs;
+
+function getFirstWorker() {
+  return app.workers.values().next().value;
+}
+
+exports["test load worker"] = function(assert, done) {
+  function onReady(tab) {
+    if (tab.url === url) {
+      assert.equal(app.workers.size, 1, "Worker is loaded");
+    }
+    done();
+  }
+  assert.equal(app.workers.size, 0, "app.workers start at 0");
+  tabs.open({url, onReady});
+};
+
+exports["test removing worker on url change"] = function(assert, done) {
```

Comment:
there's not really a point to having a generator now since it's a single async function, it's less code/easier to read this way